### PR TITLE
Hannah and Matilda, Project Labyrinth

### PR DESCRIPTION
**Pull request by jesinkey**
Link to my deployed project:

My reflections on how this week's project turned out:
Happy with the functionality but no time for styling so it is very basic.

Things I'd like to have clarified or explained in more detail:
I think thunks bear repeating and I would have liked even more weeks with redux.

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
-import React from 'react'
+import React from 'react';
+// import { BrowserRouter, Switch, Route } from 'react-router-dom';
```
Comment (sssylvasss) [inline code/src/App.js:2]: Remove unused code:)

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+import { Provider } from 'react-redux';
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+
+// import { StartPage } from 'components/StartPage';
```
Comment (sssylvasss) [inline code/src/App.js:6]: and here remove

File: code/src/App.js [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+import { Main } from 'components/Main';
+
+import { game } from 'reducers/game';
+import { ui } from 'reducers/ui';
+// import { LoadingIndicator } from 'components/LoadingIndicator';
```
Comment (sssylvasss) [inline code/src/App.js:11]: and here...:)

File: code/src/components/GamePage.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  if (user === null) {
+    content = <StartPage />;
+  } else {
+    content = (
+      <div>
+        <h2 className='description-container nes-container'>
+          {descriptionText}
+        </h2>
+        <p className='coordinates'>Your coordinates are: {coordinates}</p>
+        <h3 classname='play-text'>Choose your direction wisely...</h3>
```
Comment (sssylvasss) [inline code/src/components/GamePage.js:25]: Hide this on the last page of the game maybe?

File: code/src/components/Main.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { LoadingIndicator } from 'components/LoadingIndicator';
+import { GamePage } from 'components/GamePage';
+
+export const Main = () => {
+  const loading = useSelector((store) => store.ui.loading);
+  console.log(loading);
```
Comment (sssylvasss) [inline code/src/components/Main.js:9]: Delete console.log

File: code/src/components/StartPage.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+// import { Link } from 'react-router-dom';
```
Comment (sssylvasss) [inline code/src/components/StartPage.js:2]: delete!!:)

File: code/src/index.css [hunk 7]
```diff
@@ -1,13 +1,82 @@
...
 code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
     monospace;
 }
+
+/* .startpage-container {
+  background-image: url('assets/labyrinth-background.jpeg');
+} */
+
```
Comment (sssylvasss) [inline code/src/index.css:17]: Clean comments

**Review by sssylvasss — APPROVED** [review]
Nice and clean! Well structured and easy to read. 

Well done!!
Ylva och Laima

File: code/src/components/GamePage.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            <p className='action-description '>{action.description}</p>
+            <button
+              className='option-button nes-btn is-primary'
+              onClick={() => dispatch(fetchNext(action.direction, action.type))}
+            >
+              <h2> Go {action.direction}</h2>
+            </button>
+          </div>
+        ))}
+      </div>
```
Comment (KarolCislowski) [inline code/src/components/GamePage.js:40]: You are returning something so it is a great inductor of "It should be a separate component" 😺

File: code/src/components/GamePage.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            key={action.direction}
+          >
+            <p className='action-description '>{action.description}</p>
+            <button
+              className='option-button nes-btn is-primary'
+              onClick={() => dispatch(fetchNext(action.direction, action.type))}
+            >
+              <h2> Go {action.direction}</h2>
+            </button>
+          </div>
```
Comment (KarolCislowski) [inline code/src/components/GamePage.js:38]: and here inside you returning again some code inside the .map() so again it should be considered a good candidate to become a separate component.

File: code/src/components/LoadingIndicator.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export const LoadingIndicator = () => {
+  return (
+    <>
+      <div>
+        <p>Loading!!!</p>
+      </div>
+    </>
```
Comment (KarolCislowski) [inline code/src/components/LoadingIndicator.js:9]: You have only one immediate child element (div) so you shouldn't use react fragment here.

File: code/src/components/GamePage.js (hunk 4 above)
Comment (KarolCislowski) [reply code/src/components/GamePage.js:25]: Yes 🤓

**Review by KarolCislowski — APPROVED** [review]
nice app 🥳

File: code/src/components/GamePage.js (hunk 8 above)
Comment (jesinkey) [reply code/src/components/GamePage.js:40]: Thanks! Will think about this in future

File: code/src/components/GamePage.js (hunk 9 above)
Comment (jesinkey) [reply code/src/components/GamePage.js:38]: Noted!

**Pull request closed without merging**